### PR TITLE
Added pCloudy support

### PR DESCRIPTION
**Pull request by anandbagmar**
- Added capability to run tests in pCloudy Device Farm (also tested in on-premise pCloudy Device Farm)
- Configuration and capability parameters can be overridden from environment variables or as System properties
- Changed System.out to LOGGER

### CODE REVIEW

**Review by SrinivasanTarget — CHANGES_REQUESTED** [review]
CI doesn't seem to be happy. Can we fix it?

File: src/main/java/com/appium/capabilities/Capabilities.java [hunk 1]
```diff
@@ -105,7 +112,7 @@ private void processJSONObject(JSONObject objectToUpdate,
                                    String keyStr,
                                    Object keyvalue) {
         currentPath.append(keyStr);
-        String getFromEnv = System.getenv(currentPath.toString());
+        String getFromEnv = Variable.getOverriddenStringValue(currentPath.toString());
```
Comment (SrinivasanTarget) [inline src/main/java/com/appium/capabilities/Capabilities.java:115]: IMO, Class Name `Variable` is not a good selection.

Comment (anandbagmar) [reply src/main/java/com/appium/capabilities/Capabilities.java:115]: Any suggestion for a better name?

Comment (anandbagmar) [reply src/main/java/com/appium/capabilities/Capabilities.java:115]: Also, didn't understand why the CI is failing. Can you point me to where and why it is complaining?

Comment (saikrishna321) [reply src/main/java/com/appium/capabilities/Capabilities.java:115]: @anandbagmar 

>>INFO [main] (HostMachineDeviceManager.java:338)- Chrome not found on device - '4618862C-1074-4F9C-804C-7E2AEE964366'
java.util.NoSuchElementException: No value present
    at java.util.OptionalInt.getAsInt(OptionalInt.java:118)

This error is fixed in master. Can you please update your branch with master

**Comment by saikrishna321** [comment]
@anandbagmar CI on master is 🟢

File: src/main/java/com/appium/capabilities/Capabilities.java (hunk 1 above)
Comment (anandbagmar) [reply src/main/java/com/appium/capabilities/Capabilities.java:115]: @saikrishna321 - I updated the latest ATD changes in my fork. Don't think it triggered the tests again though.

Comment (anandbagmar) [reply src/main/java/com/appium/capabilities/Capabilities.java]: @saikrishna321 , @SrinivasanTarget  - I have merged ATD master in the fork. Also renamed Variable class to OverriddenVariable. The tests are still failing in CI as there are no devices available for some of the pipeline jobs. Please take a look at that.

**Comment by anandbagmar** [comment]
> CI doesn't seem to be happy. Can we fix it?

Done

**Pull request closed without merging**